### PR TITLE
az replace deployments operation list

### PR DESCRIPTION
**Pull request by vhvb1989**
Replacing subscription/resourceGroup deployment operation

### CODE REVIEW

File: cli/azd/pkg/infra/azure_resource_manager_test.go [hunk 1]
```diff
@@ -119,22 +137,36 @@ func TestGetDeploymentResourceOperationsSuccess(t *testing.T) {
...
-	mockContext.CommandRunner.When(func(args exec.RunArgs, command string) bool {
-		return strings.Contains(command, "az deployment operation sub list")
-	}).RespondFn(func(args exec.RunArgs) (exec.RunResult, error) {
+	mockContext.HttpClient.When(func(request *http.Request) bool {
+		return request.Method == http.MethodGet && strings.Contains(
+			request.URL.Path,
+			"/subscriptions/SUBSCRIPTION_ID/resourcegroups/resource-group-name/deployments/group-deployment-id/operations",
+		)
+	}).RespondFn(func(request *http.Request) (*http.Response, error) {
 		subCalls++
```
Comment (wbreza) [inline cli/azd/pkg/infra/azure_resource_manager_test.go:146]: FYI - We have a couple little test helper functions to make creating mock http responses a little easier.  `mocks.CreateHttpResponseWithBody(...)` and `mocks.CreateEmptyHttpResponse(...)`.  They both handle any JSON marshelling and creating the body reader/writer.  We are using these successfully in many other tests.

File: cli/azd/pkg/tools/azcli/deployments_operations.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+	return client, nil
+}
+
+func (cli *azCli) ListSubscriptionDeploymentOperations(
+	ctx context.Context,
+	subscriptionId string,
+	deploymentName string,
+) (result []*armresources.DeploymentOperation, err error) {
+	result = make([]*armresources.DeploymentOperation, 0)
```
Comment (wbreza) [inline cli/azd/pkg/tools/azcli/deployments_operations.go:33]: If initializing with 0 entries we might as well just use `[]*armresources.DeploymentOperation{}`
```suggestion
	result = []*armresources.DeploymentOperation{}
```

File: cli/azd/pkg/tools/azcli/deployments_operations.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	}
+
+	return client, nil
+}
+
+func (cli *azCli) ListSubscriptionDeploymentOperations(
+	ctx context.Context,
+	subscriptionId string,
+	deploymentName string,
+) (result []*armresources.DeploymentOperation, err error) {
```
Comment (wbreza) [inline cli/azd/pkg/tools/azcli/deployments_operations.go:32]: Any reason to be using named outputs here when you're already returning in all the required places?

File: cli/azd/pkg/tools/azcli/deployments_operations.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+	return result, nil
+}
+
+func (cli *azCli) ListResourceGroupDeploymentOperations(
+	ctx context.Context,
+	subscriptionId string,
+	resourceGroupName string,
+	deploymentName string,
+) (result []*armresources.DeploymentOperation, err error) {
```
Comment (wbreza) [inline cli/azd/pkg/tools/azcli/deployments_operations.go:62]: Same question here regarding named outputs and new empty slice.

**Review by wbreza — APPROVED** [review]
I have a few questions & suggestions - otherwise looks great!

File: cli/azd/pkg/tools/azcli/deployments_operations.go (hunk 3 above)
Comment (vhvb1989) [reply cli/azd/pkg/tools/azcli/deployments_operations.go:32]: timing only..  I added initially to simplify the return and I also thought that it was going to init an empty array for me... but it is just a `nil` :)  .  I updated the function but forgot to remove the names from there.  I will remove them, Thanks

File: cli/azd/pkg/infra/azure_resource_manager_test.go (hunk 1 above)
Comment (vhvb1989) [reply cli/azd/pkg/infra/azure_resource_manager_test.go:146]: uff, that would've been easier :) .  I had to type strings :)

I'll use it for the last API that I need to do.

And hopefully in the future we don't have to keep manually creating mocks :)

**Pull request merged**